mmv1/third_party/terraform/services/cloudrunv2/resource_cloud_run_v2_service_test.go.erb

```diff
@@ -666,3 +667,79 @@ resource "google_cloud_run_v2_service" "default" {
 }
 `, context)
 }
+
+<% unless version == "ga" -%>
+func TestAccCloudRunV2Service_cloudrunv2ServiceCustomAudienceUpdate(t *testing.T) {
+	t.Parallel()
+
+  serviceName := fmt.Sprintf("tf-test-cloudrun-service%s", acctest.RandString(t, 10))
+	context := map[string]interface{}{
+		"service_name": serviceName,
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
```

Comment:
I think you might need to make the changes described in https://googlecloudplatform.github.io/magic-modules/develop/test/#add-an-update-test for beta tests - but no need if it isn't complaining about the provider used for tests I suppose.

Reply:
If you don't specify the `launch_stage` you get this back from the API, I did however skip over a test on how to create a test for the BETA provider. I will make a few changes.

```
Error: Error creating Service: googleapi: Error 400: The feature 'Custom audiences' is not supported in the declared launch stage on resource cloudrun-service. The launch stage annotation should be specified at least as BETA. Please visit https://cloud.google.com/run/docs/troubleshooting#launch-stage-validation for in-depth troubleshooting documentation.
│ 
│   with google_cloud_run_v2_service.default,
│   on main.tf line 58, in resource "google_cloud_run_v2_service" "default":
│   58: resource "google_cloud_run_v2_service" "default" {
```

Reply:
That said this issue might be related, https://github.com/hashicorp/terraform-provider-google/issues/13748 / b/269878160

Reply:
@melinath Can we re-run the tests?

That said, with this solution we first set the custom audience in the initial creation and then remove it to satisfy the update test condition.

Will have to ignore the import state with the inconsistent `launch_stage` field from the API.

```
➜  terraform-provider-google-beta git:(main) ✗ make testacc TEST=./google-beta/services/cloudrunv2  TESTARGS='-run=TestAccCloudRunV2Service_cloudrunv2ServiceCustomAudienceUpdate'
==> Checking that code complies with gofmt requirements...
go vet
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google-beta/services/cloudrunv2 -v -run=TestAccCloudRunV2Service_cloudrunv2ServiceCustomAudienceUpdate -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google-beta/version.ProviderVersion=acc"
=== RUN   TestAccCloudRunV2Service_cloudrunv2ServiceCustomAudienceUpdate
=== PAUSE TestAccCloudRunV2Service_cloudrunv2ServiceCustomAudienceUpdate
=== CONT  TestAccCloudRunV2Service_cloudrunv2ServiceCustomAudienceUpdate
--- PASS: TestAccCloudRunV2Service_cloudrunv2ServiceCustomAudienceUpdate (50.55s)
PASS
ok      github.com/hashicorp/terraform-provider-google-beta/google-beta/services/cloudrunv2     51.217s

```